File: dayFour/Java/DayFourTask.java
import java.util.Arrays;

public class DayFourTask{
    public static int[] sumOfTwoNumbers(int[] array, int number){
        int[] newArray = new int[2];
        int count = 0;

        for(int index = 0; index < array.length; index++){
            for(count = 0; count < array.length; count++){
                if(array[index] + array[count] == number){
                    newArray[0] = array[index];
                    newArray[1] = array[count];
                }
            }
            if(newArray[0] == array[index] && newArray[1] == array[count]){
                break;
            }
        }
        return newArray;
    }

    public static int[] rangeBetweenLargestAndSmallestNumbers(int[] array){
        int largest = array[0];

        for(int index = 1; index < array.length; index++){
            if(array[index] > largest){
                largest = array[index];
            }
        }
        int smallest = array[0];

        for(int index = 1; index < array.length; index++){
            if(array[index] < smallest){
                smallest = array[index];
            }
        }

        int[] range = new int[(largest - smallest) + 1];

        for(int index = 0; index < range.length; index++){
            range[index] = smallest;
            smallest++;
        }

        return range;
    }

//    public static String[] consonantOrVowelsInWords(String[] array){
//
//
//    }

    public static void main(String[] args){
        int[] taskOne = {8, 6, 12, 4, -2};
//        System.out.println(Arrays.toString(sumOfTwoNumbers(taskOne, 6)));

        int[] taskThree = {14, 9, 6, 5, 8, 10};

        System.out.println(Arrays.toString(rangeBetweenLargestAndSmallestNumbers(taskThree)));
    }
}

File: dayFour/JavaScript/day-four-task.js
function sumOfTwoNumbers(array, number){
    let newArray = [];
    let count = 0;

    for(let index = 0; index < array.length; index++){
        for(count = 0; count < array.length; count++){
            if(array[index] + array[count] == number){
                newArray[0] = array[index];
                newArray[1] = array[count];
                break;
            }
        }
        if(newArray[0] == array[index] && newArray[1] == array[count]){
            break;
        }
    }

    return newArray;

}

function rangeBetweenLargestAndSmallestNumbers(array){
    let largest = array[0];

    for(let index = 1; index < array.length; index++){
        if(array[index] > largest){
            largest = array[index];
        }
    }
    let smallest = array[0];

    for(let index = 1; index < array.length; index++){
        if(array[index] < smallest){
            smallest = array[index];
        }
    }

    let range = [];

    for(let index = 0; index < (largest-smallest) + 1; index++){
        range[index] = index;
    }



    for(let index = 0; index < range.length; index++){
        range[index] = smallest;
        smallest++;
    }

    return range;
}

//console.log(sumOfTwoNumbers([8, 6, 12, 4, -2], 6))
//console.log(rangeBetweenLargestAndSmallestNumbers([14, 9, 6, 5, 8, 10]))

module.exports = { sumOfTwoNumbers, rangeBetweenLargestAndSmallestNumbers }
